packages/kinvey-datastore/src/repositories/repository-provider.js

```diff
@@ -0,0 +1,47 @@
+import Client from 'kinvey-client';
+import { InmemoryOfflineRepository } from './offline-repositories';
+import { NetworkRepository } from './network-repository';
+import { storageType } from './storage-type';
+import { persisterProvider } from '../persisters';
+import { InmemoryCrudQueue, ensureArray } from '../utils';
+
+const defaultConstructorName = 'default';
+const inmemoryRepoBuilder = (queue) => {
+  const persister = persisterProvider.getPersister();
+  return new InmemoryOfflineRepository(persister, queue);
+};
+
+// TODO: all inmemory instances should share the queue. are there better ways to share it?
+// queue needs to be by collection
+const queue = new InmemoryCrudQueue();
+// TODO: not great to do this here, but tests fail otherwise
+let supportedStorages = {
+  [storageType.inmemory]: inmemoryRepoBuilder,
+  [defaultConstructorName]: inmemoryRepoBuilder
+};
+
+function getRepoType() {
```

Comment:
Seems like a redundant method. All you are doing is returning the `storageType` from `Client`. The `getOfflineRepository` method below can directly call that.

Reply:
This is the last thing I was working on before we decided to start moving code. It's the part I mentioned needs to be polished. I wasn't/am not very sure what the best way to achieve this is yet, so I'm separating actions into functions. I don't think leaving it this way is too bad either, though. Other functions not caring where the type comes from, naming the operation, bridging setting name and actual logic. Will improve during development.

Reply:
I didn't fully understand your comment, but I'll wait for the improvements and review again :)

Reply:
Deal :)